internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -216,6 +218,29 @@ func (c *Config) KexecLoad(r runtime.Runtime, disk string) error {
 			return fmt.Errorf("failed to read cmdline from uki: %w", err)
 		}
 
+		smbiosInfo, err := smbiosinternal.GetSMBIOSInfo()
+		if err == nil {
+			for _, structure := range smbiosInfo.Structures {
+				if structure.Header.Type != 11 {
+					continue
+				}
+
+				if structure.Strings != nil {
```

Comment:
this should be shorter:

```suggestion
        const kernelCmdlineExtra = "io.systemd.stub.kernel-cmdline-extra="
	for _, s := range structure.Strings {
            if strings.HasPrefix(s, kernelCmdlineExtra) {
                cmdline.WriteString(" " + s[len(kernelCmdlineExtra):])
            }
        }
```